optimum/exporters/openvino/model_configs.py

```diff
@@ -0,0 +1,266 @@
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from copy import deepcopy
+from typing import TYPE_CHECKING, Any, Dict, Optional, Tuple, Union
+
+from packaging import version
+from transformers.utils import is_tf_available
+
+from optimum.exporters.onnx.config import TextDecoderOnnxConfig, TextDecoderWithPositionIdsOnnxConfig
+from optimum.exporters.openvino.model_patcher import ChatGLMModelPatcher, MixtralModelPatcher
+from optimum.exporters.tasks import TasksManager
+from optimum.utils import DEFAULT_DUMMY_SHAPES
+from optimum.utils.input_generators import (
+    DummyInputGenerator,
+    DummyPastKeyValuesGenerator,
+    DummyTextInputGenerator,
+    MistralDummyPastKeyValuesGenerator,
+)
+from optimum.utils.normalized_config import NormalizedTextConfig
+
+
+def init_model_configs():
+    supported_model_types = [
+        "_SUPPORTED_MODEL_TYPE",
+        "_DIFFUSERS_SUPPORTED_MODEL_TYPE",
+        "_TIMM_SUPPORTED_MODEL_TYPE",
+        "_SENTENCE_TRANSFORMERS_SUPPORTED_MODEL_TYPE",
+    ]
+
+    for supported_models_config in supported_model_types:
+        supported_models = getattr(TasksManager, supported_models_config)
+        for model, export_configs in supported_models.items():
+            if "onnx" not in export_configs:
+                continue
+            onnx_config = export_configs["onnx"]
+            supported_models[model]["openvino"] = deepcopy(onnx_config)
+
+        setattr(TasksManager, supported_models_config, supported_models)
+
+
+init_model_configs()
+
+
+if TYPE_CHECKING:
+    from transformers.modeling_utils import PreTrainedModel
+
+    from optimum.exporters.onnx.model_patcher import ModelPatcher
+
+    if is_tf_available():
+        from transformers.modeling_tf_utils import TFPreTrainedModel
+
+
+register_in_tasks_manager = TasksManager.create_register("openvino", overwrite_existing=True)
+
+
+@register_in_tasks_manager("baichuan", *["text-generation", "text-generation-with-past"])
+class BaichaunOpenVINOConfig(TextDecoderOnnxConfig):
+    DEFAULT_ONNX_OPSET = 13
+    NORMALIZED_CONFIG_CLASS = NormalizedTextConfig.with_args(
+        num_layers="num_hidden_layers", num_attention_heads="num_attention_heads", hidden_size="hidden_size"
+    )
+
+
+@register_in_tasks_manager("jais", *["text-generation", "text-generation-with-past"])
+class JaisOpenVINOConfig(TextDecoderOnnxConfig):
+    DEFAULT_ONNX_OPSET = 13
+    NORMALIZED_CONFIG_CLASS = NormalizedTextConfig.with_args(
+        num_layers="n_layer", num_attention_heads="n_head", hidden_size="n_embd"
+    )
+
+
+@register_in_tasks_manager("qwen2", *["text-generation", "text-generation-with-past"])
+class Qwen2OpenVINOConfig(TextDecoderWithPositionIdsOnnxConfig):
```

Comment:
for all configs inheriting from `TextDecoderWithPositionIdsOnnxConfig`, is it optional or is the exported model's outputs  incorrect for `batch_size>1` ? If optional I would prefer to remove them for now (to keep consistency as as that's we do for all models where it's optional)

Reply:
yes, all of them inherit this issue from llama